### PR TITLE
[DOC] Add badges to the change log

### PR DESCRIPTION
**Pull request by ymzayek**
Supersedes and closes #3280.

Opinions encouraged on which labels to use!
For now I just put API, Plotting, Doc, Deprecation, Test

I'm using inline badges formatted by sphinx design, from which our theme comes (https://sphinx-design.readthedocs.io/en/furo-theme/badges_buttons.html). It seems possible to create custom ones however I wouldn't do this unless we feel it's necessary to cover all desired labels. For now, what's there seems good and plenty enough.

Thanks @NicolasGensollen for initiating this!

### CODE REVIEW

File: doc/changes/0.10.0.rst
Comment (tsalo) [inline doc/changes/0.10.0.rst:58]: Maybe Doc instead of DOC, since it's not an acronym/initialism.

File: doc/changes/0.10.0.rst
Comment (bthirion) [inline doc/changes/0.10.0.rst:64]: No badge here ?

File: doc/changes/latest.rst
Comment (bthirion) [inline doc/changes/latest.rst:53]: and here ?

**Comment by ymzayek** [comment]
Badge mapping so far (subject to change):

- :bdg-primary:\`Doc\`
- :bdg-secondary:\`Maint\`
- :bdg-success:\`API\`
- :bdg-info:\`Plotting\`
- :bdg-warning:\`Test\`
- :bdg-danger:\`Deprecation\`
- :bdg-light:\`Code\`

Not assigned:

- :bdg-dark:``

File: doc/changes/0.10.0.rst
Comment (ymzayek) [reply doc/changes/0.10.0.rst]: Added now

File: doc/changes/latest.rst
Comment (ymzayek) [reply doc/changes/latest.rst]: Same

**Comment by ymzayek** [comment]
I think I might apply them backwards just to v0.9, and then open a PR that would be a `Good first issue` for contributors to apply them on the other versions. If that sounds good to everybody

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Comment by ymzayek** [comment]
This one is ready. Once merged I'll open an issue to apply the badges backwards to the rest of the changelog.

**Review by Remi-Gau — APPROVED** [review]

**Comment by ymzayek** [comment]
@Remi-Gau I would say this is something that maintainers can do before a release (moving forward). In that case we can document it in the maintenance docs in the [making a release section](https://nilearn.github.io/stable/maintenance.html#how-to-make-a-release).

For finishing applying it backwards, I'll open an issue.

WDYT?

**Comment by Remi-Gau** [comment]
> @Remi-Gau I would say this is something that maintainers can do before a release (moving forward). In that case we can document it in the maintenance docs in the [making a release section](https://nilearn.github.io/stable/maintenance.html#how-to-make-a-release).
> 
> For finishing applying it backwards, I'll open an issue.
> 
> WDYT?

yup that's good with me

**Comment by bthirion** [comment]
Merging, then.

**Pull request merged**
